lib/Onyx.js

```diff
@@ -1571,7 +1593,99 @@ function update(data) {
         }
     });
 
-    return clearPromise.then(() => Promise.all(_.map(promises, (p) => p())));
+    return clearPromise.then(() =>
+        multiGet(_.keys(updateQueue)).then((pairs) => {
```

Comment:
After a closer look, don't we eventually read all the values from cache/storage anyway before updating the cached values?

https://github.com/Expensify/react-native-onyx/blob/3f77c6bd2030979da3d581ea01fc9fd5e6d646e5/lib/Onyx.js#L1525-L1530

Reply:
Since we need to wait for all the Storage write operations to complete, and then perform the `cache.get()` for each key and merge in-memory anyway to update the cache, is there any benefit from the `JSON_PATCH` in the current implementation?

I see there already was a similar discussion [here](https://github.com/Expensify/react-native-onyx/pull/238#discussion_r1123966066), looks like it was not completely resolved 🤔

Reply:
Yes, we do `get` the latest value and perform a full merge with it to update cache, but we only try to send the delta for the write to storage.
The only benefit seems is we're sparing sending the full data for the write operation (and in fact this is only on native on web we send the full data for the merge)